config/routes.rb

```diff
@@ -77,7 +77,11 @@
         json_api_resources :pages, controller: "project_pages"
       end
 
-      json_api_resources :workflows, links: [:subject_sets], versioned: true
+      json_api_resources :workflows, links: [:subject_sets], versioned: true do
+        member do
+          post :retire_subject
```

Comment:
I agree with that, that's also the way I'd reason about it. I assume the database approaches this differently because a workflow has many subjects and the queries might get too slow? Is that also the reason behind the current array-of-ids vs a join table for `SubjectWorkflowRetirements`?